src/Cake.Core/Configuration/Parser/ConfigurationParser.cs

```diff
@@ -37,7 +37,7 @@ public ConfigurationParser(IFileSystem fileSystem, ICakeEnvironment environment)
             using (var stream = file.OpenRead())
             using (var reader = new StreamReader(stream))
             {
-                return Read(reader.ReadToEnd());
+                return Read(Environment.ExpandEnvironmentVariables(reader.ReadToEnd()));
```

Comment:
The scope of the expansion here seems a little too broad, and I'm not 100% sure if it will work for expanding multiple variables on all platforms, considering different line-endings, and the tests don't cover that.

With the current scope, the following scenarios would be possible:

```ini
[Section]
%A%=%B%
```

or even

```ini
[%A%]
%B%=%C%
```

I'm not sure we'd want to support expansion on the whole file rather than on invididual `values` of each field... We probably only want this one scenario to work:

```ini
[Section]
Key=%VALUE%
```

@cake-build/cake-team What do you think?

I think it would be safer to perform the expansion [directly on each value](https://github.com/cake-build/cake/blob/04a98f8e25f4f676f99edb9aef9ac3e724eb809d/src/Cake.Core/Configuration/Parser/ConfigurationParser.cs#L59) as we parse individual entries:

```csharp
case ConfigurationTokenKind.Value:
     var pair = ParseKeyAndValue(tokens, section);
     result[pair.Key] = _environment.ExpandEnvironmentVariables(pair.Value);
     break;
```

The other good thing is that the [`ConfigurationParser` already has access to `ICakeEnvironment`](ExpandEnvironmentVariables), so we can use that instead of calling `Environment` directly.

Reply:
Ok, that's reasonable and should cover my needs.
Should be fairly easy to implement.

Reply:
Done, now changed to using environment variable substitution for values only.  
Sorry about the force-push, I am not used to the workflow for pullrequests on Github repositories.